### ISSUE TITLE
[Console Editor] Surface "Run Selected Code" feature

### ISSUE THREAD

**Issue by nchevobbe** [firefox-devtools/ux#91]
Since you can select only what's currently selected on the input, we might want to reflect that in the UI to indicate to the user that we'll do a specific action.

Ideas so far:
- Changing the run button text to Run Selected Code
- Changing the run button tooltip
- Changing the run button style (e.g. background-color could match the background color of selected text)
- Adding an inline icon after the selection

---

You can see how some of those might look in [Bug 1577115](https://bugzilla.mozilla.org/show_bug.cgi?id=1577115)

**Comment by juliandescottes** [comment]
Another idea:
- the `Run` button always runs the whole code, but we add a context menu option on the selected text "Run the selected text"

**Comment by fvsch** [comment]
Oooh are we planning Cmd+Shift+Enter to run the selected text maybe?

**Comment by nchevobbe** [comment]
Actually this did come up in a discussion I had with Irakli in Whistler.
He was saying it was very handy when you're working on a big script ( = you want to debug only a specific part of it without wiping out everything you wrote).
A specific shortcut sounds good to me + maybe some hint, somewhere (still don't know where :D )

**Comment by fvsch** [comment]
>  A specific shortcut sounds good to me + maybe some hint, somewhere (still don't know where :D )

We could:
- tell about this shortcut in the welcome message
- treat the welcome message as a kind of mini-help, and let users show it again via a "(?)" button in the toolbar

**Comment by violasong** [comment]
Maybe an easy option for now is: If you select some code and hit cmd-enter or click the run button, the output starts with the explanatory line: "Running selected code:"

**Comment by violasong** [comment]
I do like the idea of adding this to the welcome message, if we want to encourage its usage.

**Comment by digitarald** [comment]
> I do like the idea of adding this to the welcome message, if we want to encourage its usage.

Yes. Having some visual feedback when selecting text that execution mode changes (or a new one is available) would be great to encourage discovery (for anybody who did not read the onboarding, especially).
